components/core/src/clp_s/clp-s.cpp

```diff
@@ -131,6 +159,157 @@ bool compress(CommandLineArguments const& command_line_arguments) {
+    if (false == in_file.is_open()) {
+        SPDLOG_ERROR("Failed to open input file: {}", path);
+        return false;
+    }
+    std::filesystem::path input_path{path};
+    std::string filename = input_path.filename().string();
+    std::string out_path = option.irs_dir + "/" + filename + ".ir";
+
+    clp_s::FileWriter out_file;
+    out_file.open(out_path, clp_s::FileWriter::OpenMode::CreateForWriting);
+    clp_s::ZstdCompressor zc;
+    try {
+        zc.open(out_file, option.compression_level);
+    } catch (clp_s::ZstdCompressor::OperationFailed& error) {
+        SPDLOG_ERROR("Failed to open ZSTDcompressor - {}", error.what());
+        in_file.close();
+        out_file.close();
+        return false;
+    }
+
+    std::string line = "";
+    size_t total_size = 0;
+
+    if (in_file.is_open()) {
+        while (getline(in_file, line)) {
+            try {
+                auto j_obj = nlohmann::json::parse(line);
+                if (false
+                    == unpack_and_serialize_msgpack_bytes(
+                            nlohmann::json::to_msgpack(j_obj),
+                            serializer
+                    ))
+                {
+                    SPDLOG_ERROR("Failed to serialize msgpack bytes for line: {}", line);
+                    in_file.close();
+                    out_file.close();
+                    zc.close();
+                    return false;
+                }
+                auto bufferSize = serializer.get_ir_buf_view().size();
+                if (bufferSize >= option.max_ir_buffer_size) {
+                    total_size = total_size + bufferSize;
+                    zc.write(
+                            reinterpret_cast<char*>(
+                                    const_cast<int8_t*>(serializer.get_ir_buf_view().data())
+                            ),
+                            bufferSize
+                    );
+                    zc.flush();
+                    serializer.clear_ir_buf();
+                }
+            } catch (nlohmann::json::parse_error const& e) {
+                SPDLOG_ERROR("JSON parsing error: {}", e.what());
+                in_file.close();
+                out_file.close();
+                zc.close();
+                return false;
+            } catch (std::exception const& e) {
+                SPDLOG_ERROR("Error during serialization: {}", e.what());
+                in_file.close();
+                out_file.close();
+                zc.close();
+                return false;
+            }
+        }
+        total_size = total_size + serializer.get_ir_buf_view().size();
+        zc.write(
+                reinterpret_cast<char*>(const_cast<int8_t*>(serializer.get_ir_buf_view().data())),
+                serializer.get_ir_buf_view().size()
+        );
+        std::vector<int8_t> ir_buf;
+        ir_buf.push_back(clp::ffi::ir_stream::cProtocol::Eof);
+        zc.write(reinterpret_cast<char*>(ir_buf.data()), ir_buf.size());
```

Comment:
```suggestion
                constexpr std::array<int8_t, 1> cEndOfStreamBuf{clp::ffi::ir_stream::cProtocol::Eof};
        zc.write(
                clp::size_checked_pointer_cast<char const>(cEndOfStreamBuf.data()),
                cEndOfStreamBuf.size()
        );
```
We can make it as a compile-time constant.